tsl/src/remote/copy_fetcher.c

```diff
@@ -449,7 +454,9 @@ copy_fetcher_complete(CopyFetcher *fetcher)
 				 * nor the expected number of columns. This provides an extra
 				 * check against somehow getting out of sync with the data.
 				 */
+#ifdef USE_ASSERT_CHECKING
 				fetcher->file_trailer_received = true;
```

Comment:
Below you have a comment saying that the loop will not be executed in some cases, so maybe we can not execute it always?

The reasons we needed this flag is that the final iteration of the loop was supposed to check for EOF. Now that you are doing this check in place, we don't need the final iteration at all, right?

Reply:
This is correct. The final loop should check for EOF but it was not always executed. Now, we perform the read operation directly here and update the EOF state accordingly. I updated the comment to make this easier to understand.